perfkitbenchmarker/providers/ibmcloud/utils.py

```diff
@@ -0,0 +1,445 @@
+        logging.error('Failed to retrieve %s, no data returned', items)
+        counter += 1
+        time.sleep(10)  # 5 still fails when accounts >= 10
+    return resourceid
+
+  def ListSubnetsExtra(self):
+    """returns a list of extra subnets """
+    items = 'subnets'
+    prefix = self.flags['prefix']
+    zone = self.flags['zone']
+    logging.info('Looking up extra subnets for zone: %s', zone)
+    data_mgr = self.gen_subnetmgr
+    subnets = {}
+    counter = 1
+    while counter < 4:
+      resp = data_mgr.List()
+      if resp and items in resp:
+        for item in resp[items]:
+          itemid = item.get('id')
+          name = item.get('name')
+          ipv4_cidr_block = item.get('ipv4_cidr_block')
+          if itemid and name and name.startswith(prefix):  # extra subnets start with 'sxs'
+            if zone == item.get('zone')['name']:
+              logging.info('Found extra subnet, name: %s, id: %s', name, itemid)
+              names = name.split(DELIMITER)
+              if names[0] not in subnets:
+                subnets[names[0]] = {}
+              subnets[names[0]]['id'] = itemid
+              subnets[names[0]]['ipv4_cidr_block'] = ipv4_cidr_block  # needed to create route later
+        break  # empty or done
+      else:
+        logging.error('Failed to retrieve extra subnets, no data returned')
+        counter += 1
+        time.sleep(10)  # 5 still fails when accounts >= 10
+    return subnets
+
+  def CreateVolume(self):
+    """Creates an external disk volume """
+    kwargs = {}
+    kwargs['name'] = self.flags['name']
+    kwargs['capacity'] = self.flags['capacity']
+    kwargs['iops'] = self.flags['iops']
+    kwargs['profile'] = self.flags['profile']
+    if 'encryption_key' in self.flags:
+      kwargs['encryption_key'] = self.flags['encryption_key']
+    if 'resource_group' in self.flags:
+      kwargs['resource_group'] = self.flags['resource_group']
+    return json.dumps(self.gen_volumemgr.Create(self.flags['zone'], **kwargs))
+
+  def DeleteVolume(self):
+    """Deletes volume """
+    return self.gen_volumemgr.Delete(self.flags['volume'])
+
+  def ShowVolume(self):
+    """Shows volume """
+    return json.dumps(self.gen_volumemgr.Show(self.flags['volume']), indent=2)
+
+  def CreateVpc(self):
+    """Creastes a vpc """
+    vpcid = None
+    resp = self.gen_vpcmgr.Create(None, self.flags['name'])
+    if resp:
+      vpcid = resp.get('id')
+      self.sgid = resp.get('default_security_group')['id']
+      logging.info('Created vpc: %s, sgid: %s', vpcid, self.sgid)
+      self.flags['sgid'] = self.sgid
+      self.CreateSgRules()
+      logging.info('Created sg rules for icmp, tcp 22/443, tcp/udp 1024-50000')
+    return vpcid
+
+  def CreateKey(self):
+    """Creastes a ssh key in RIAS """
+    kwargs = {}
+    kwargs['name'] = self.flags['name']
+    resp = self.gen_keymgr.Create(self.flags['pubkey'], 'rsa', **kwargs)
+    return resp.get('id') if resp else None
+
+  def CreateInstance(self):
```

Comment:
Utils should not create any instance or resource creation. Just helper functions that are used in multiple places and CLI use. See gcp_utils, aws_utils, or azure_utils.

Reply:
Recommend splitting VM information into ibm_vm

Reply:
ok, makes sense, I will.